src/server/AppManager.ts

```diff
@@ -809,10 +810,14 @@ export class AppManager {
                 this.getAccessorManager().getEnvironmentRead(storageItem.id),
                 this.getAccessorManager().getConfigurationModify(storageItem.id)) as boolean;
 
-            await app.setStatus(isManual ? AppStatus.MANUALLY_ENABLED : AppStatus.AUTO_ENABLED, silenceStatus);
+            if (enable) {
+                status = isManual ? AppStatus.MANUALLY_ENABLED : AppStatus.AUTO_ENABLED;
+            } else {
+                status = AppStatus.DISABLED;
+                console.warn(`The App (${ app.getID() }) is disabled by itslef. Check its "onEnable" implementation for details.`);
```

Comment:
We need to avoid using `console` directly in the Engine, as it makes it harder for the server to control logging from there